File: Practica-1/S3/CryptoFilters/FilterInvestment.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   FilterInvestment.cpp
 * Author: reko98
 * 
 * Created on 25 de marzo de 2021, 12:56
 */

#include "FilterInvestment.h"

FilterInvestment::FilterInvestment() {
}

FilterInvestment::FilterInvestment(const FilterInvestment& orig) {
}

FilterInvestment::~FilterInvestment() {
}

FilterInvestment::FilterInvestment(double newValue){
    setCantidadInversion(newValue);
    
}

double FilterInvestment::apply(CryptoCurrency& moneda) {
	double valorPorcentaje;
	double cantidadIncremento;
	valorPorcentaje = (this->cantidadInversion*100)/moneda.getCapital();

	valorPorcentaje = valorPorcentaje/100;
        
	cantidadIncremento=valorPorcentaje*moneda.getValorActual();

	return moneda.getValorActual()+cantidadIncremento;
	
}

File: Practica-2/Entrega/crypto_visionary/lib/FIlterInvestment.dart
import 'package:crypto_visionary/Cryptocurrency.dart';
import 'package:flutter/material.dart';
import 'Filter.dart';

class FilterInvestment implements Filter {

  double cantidadInversion;

  FilterInvestment(double inversion) {
    cantidadInversion = inversion;
  }

  void setCatindadInversion(double nuevaCantidad){
    this.cantidadInversion = nuevaCantidad;
  }

  double getCantidadInversion(){
    return this.cantidadInversion;
  }

  @override
  double apply(Cryptocurrency moneda){
    double valorPorcentaje;
    double cantidadIncremento;
    double diferencia;

    diferencia = (cantidadInversion+moneda.getCapital()) - moneda.getCapital();
    valorPorcentaje = (diferencia/moneda.getCapital()) * 100;
    moneda.setCapital(moneda.getCapital()+cantidadInversion);

    cantidadIncremento=valorPorcentaje*moneda.getValorActual();
    return moneda.getValorActual()+cantidadIncremento;
  }




}